### PR TITLE
flip default force_consistency_checks to true

### PR DESCRIPTION
**Pull request by ajkr**
better late than never...

### CODE REVIEW

File: HISTORY.md [hunk 1]
```diff
@@ -9,11 +9,16 @@
 * Fix a bug caused by not including user timestamp in MultiGet LookupKey construction. This can lead to wrong query result since the trailing bytes of a user key, if not shorter than timestamp, will be mistaken for user timestamp.
 * Fix a bug caused by using wrong compare function when sorting the input keys of MultiGet with timestamps.
 * Upgraded version of bzip library (1.0.6 -> 1.0.8) used with RocksJava to address potential vulnerabilities if an attacker can manipulate compressed data saved and loaded by RocksDB (not normal). See issue #6703.
+* Fix a bug when making options.bottommost_compression, options.compression_opts and options.bottommost_compression_opts dynamically changeable: the modified values are not written to option files or returned back to users when being queried.
+* Fix a bug where index key comparisons were unaccounted in `PerfContext::user_key_comparison_count` for lookups in files written with `format_version >= 3`.
+* Fix many bloom.filter statistics not being updated in batch MultiGet.
+* Fix `force_consistency_checks=true` use cases to return an error if a file is wrongly deleted.
```
Comment (zhichao-cao) [inline HISTORY.md:15]: Maybe better to point out the bug swallows the returned status from CheckConsistencyForDeletes.

Comment (ajkr) [reply HISTORY.md]: done, thanks for the review!

File: db/version_builder.cc [hunk 2]
```diff
@@ -463,7 +463,10 @@ class VersionBuilder::Rep {
       const auto number = del_file.second;
       if (level < num_levels_) {
         levels_[level].deleted_files.insert(number);
-        CheckConsistencyForDeletes(edit, number, level);
+        s = CheckConsistencyForDeletes(edit, number, level);
+        if (!s.ok()) {
+          return s;
+        }
```
Comment (siying) [inline db/version_builder.cc:469]: Ideally we should have a unit test to cover this code path.

Comment (ajkr) [reply db/version_builder.cc]: Done.

**Comment by ajkr** [comment]
> I think we should separate the CheckConsistencyForDeletes() fix and patch fix 6.9.

Will submit a separate PR for that. This PR is updated to only change the default value of `force_consistency_checks`.

**Comment by ajkr** [comment]
After talking to Shrikanth we think this is too risky to cause crash loop on false positive consistency check (or even true positives that don't matter to the user). We need a way to build confidence that it works right on good DBs first.

**Pull request closed without merging**